main.js

```diff
@@ -2,14 +2,19 @@ const { join } = require('node:path');
 const { readFileSync } = require('node:fs');
 
 const lockfile = join(process.env.GITHUB_WORKSPACE, 'package-lock.json');
-const matcher = /.+\/(?<repo>[a-z-]+\/[a-z-]+)\.git#(?<hash>[0-9a-f]+)/;
+const matcher = /.+\/(?<repo>[\w-]+\/[a-z-]+)\.git#(?<hash>[0-9a-f]+)/;
 
 function getBody() {
   const { packages } = JSON.parse(readFileSync(lockfile, 'utf8'));
   return Object.entries(packages).slice(1).map(([key, value]) => {
     if (value.resolved.startsWith('git')) {
-      const { repo, hash } = matcher.exec(value.resolved).groups;
-      return `- [${repo}@\`${hash}\`](https://github.com/${repo}/commit/${hash})`;
+      const match = matcher.exec(value.resolved)
+      if (match) {
```

Comment:
I'd rather have the action fail if the regex doesn't match.
That way I'll know it needs to be fixed.

Reply:
Why? The whole `getBody` is only a gimmick. The users shouldn't worry when the regex doesn't handle their use case. This regex assumes that all repos are hosted on Github. It is expected from the beginning that it will not handle all cases.

For me the assumption, regex parsing succeeds -> can make assumption on validity on link vs a general URL

Reply:
Before I updated the regex, my idea was to try...catch the whole `getBody` part:
- let action succeed
- put error message in body, with ask to report upstream
Failing cron actions unfortunately get unnoticed often

Reply:
OK, let's print a warning in the `else` block then.

Reply:
![image](https://github.com/tree-sitter/parser-update-action/assets/7189118/2bcc43a8-7424-4aa8-9140-f57f68f3d575)

The warning is not really prominently visible. Would be also fine by failing the action with a clear error message while hoping that git repos that use would fit the regex (we already know know that you can have a URL that doesn't match the regex)